### PR TITLE
Demonstrate mailqueue recovery failure upon network outage

### PR DESCRIPTION
**Pull request by chibenwa**
Spoiler : dequeue fails

### CODE REVIEW

**Comment by chibenwa** [comment]
Fails as expected:

```
[53597cd8e20454150c9b1e1ddb41cc38b0810446] [ERROR] Errors: 
[53597cd8e20454150c9b1e1ddb41cc38b0810446] [ERROR]   RabbitMQMailQueueTest.dequeueShouldWorkAfterNetworkOutage » Completion java.io...
```

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 1]
```diff
@@ -170,6 +170,32 @@ void browseShouldReturnCurrentlyEnqueuedMailFromAllSlices() throws Exception {
...
+    @Test
+    void dequeueShouldWorkAfterNetworkOutage() throws Exception {
+        String name = "myMail";
+        getMailQueue().enQueue(defaultMail()
+            .name(name)
+            .build());
+
+        rabbitMQExtension.getRabbitMQ().pause();
+
+        new Thread(() -> {
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: Please always name threads so we can debug them once they leak.
Also, an executor seems more relevant than a thread here.

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 2]
```diff
@@ -170,6 +170,32 @@ void browseShouldReturnCurrentlyEnqueuedMailFromAllSlices() throws Exception {
...
+            try {
+                int threeMinutes = 180000;
+                Thread.sleep(threeMinutes);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            rabbitMQExtension.getRabbitMQ().unpause();
+        }).start();
+
+        getMailQueue().enQueue(defaultMail().build());
```
Comment (mbaechler) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: reading the test name, I would expect the enqueue to happen before the outage

**Comment by openpaas-james** [comment]
Build finished. No test results found.

**Comment by chibenwa** [comment]
No longer relevant after @remk recent work

**Pull request closed without merging**